ceci/stage.py

```diff
@@ -811,69 +906,86 @@ def read_config(self, args):
         # This is just the config info in the file for this stage.
         # It may be incomplete - there may be things specified on the
         # command line instead, or just using their default values
-        stage_config = overall_config.get(self.name, {})
+        stage_config = overall_config.get(self.instance_name, {})
         input_config.update(stage_config)
 
-        # Here we build up the actual configuration we use on this
-        # run from all these sources
-        my_config = {}
-
-        # Loop over the options of the pipeline stage
-        for x in self.config_options:
-            opt = None
-            opt_type = None
-
-            # First look for a default value,
-            # if a type (like int) is provided as the default it indicates that
-            # this option doesn't have a default (i.e. is mandatory) and should
-            # be explicitly provided with the specified type
-            if type(self.config_options[x]) is type:
-                opt_type = self.config_options[x]
-
-            elif type(self.config_options[x]) is list:
-                v = self.config_options[x][0]
-                if type(v) is type:
-                    opt_type = v
-                else:
-                    opt = self.config_options[x]
-                    opt_type = type(v)
-            else:
-                opt = self.config_options[x]
-                opt_type = type(opt)
-
-            # Second, look for the option in the configuration file and override
-            # default if provided TODO: Check types
-            if x in input_config:
-                opt = input_config[x]
-
-            # Finally check for command line option that would override the value
-            # in the configuration file. Note that the argument parser should
-            # already have taken care of type
-            if args[x] is not None:
-                opt = args[x]
-
-            # Finally, check that we got at least some value for this option
-            if opt is None:
-                raise ValueError(
-                    f"Missing configuration option {x} for stage {self.name}"
-                )
+        self._configs.set_config(input_config, args)
 
-            my_config[x] = opt
+    def get_config_dict(self, global_config, reduce_config=False):
```

Comment:
Could you make global_config optional please, and also maybe rename it as `ignore` or something like that - in a few places in TXPipe we assume the config is a simple dictionary right now, e.g. for passing things to other configuration mechanisms, and I'll need to replace those to use this function, and making it as simple as possible would help.

Reply:
Done.